sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -485,6 +520,113 @@ def test_fixed_windows_combine(self):
                   equal_to([('c', 3), ('c', 10), ('d', 5), ('d', 17)]),
                   label='sum per key')
 
+  # Test that three different kinds of metrics work with a customized
+  # CounterIncrememtingCombineFn.
+  def test_custormized_counters_in_combine_fn(self):
+    p = TestPipeline()
+    input = (p
+             | beam.Create([('c', 'i'),
+                            ('c', 'go'),
+                            ('c', 'run'),
+                            ('d', 'beam'),
+                            ('d', 'tests')]))
+
+    # The result of concatenating all values regardless of key.
+    global_concat = (input
+                     | beam.Values()
+                     | beam.CombineGlobally(CounterIncrememtingCombineFn())
+                     | "sort global result" >> _SortLists)
+
+    # The (key, concatenated_string) pairs for all keys.
+    concat_per_key = (input
+                      | beam.CombinePerKey(CounterIncrememtingCombineFn())
+                      | "sort per key result" >> _SortLists)
+
+    result = p.run()
+    result.wait_until_finish()
+
+    # Verify the concatenated strings are correct.
+    expected_concat_per_key = [('c', 'ginoru'), ('d', 'abeemsstt')]
```

Comment:
Nit: it's really hard to tell from reading this whether it is correct. Maybe make the input something like 

('key1': 'a'), ('key1': 'ab'), .., ('key2', 'xyz'), ...

Reply:
Done.